skbio/io/format/genbank.py

```diff
@@ -562,8 +562,7 @@ def _parse_locus(lines):
                ' +(bp|aa|rc)'
                ' +(.*DNA|.*RNA)?'
                ' +(linear|circular)?'
-               ' +(PRI|ROD|MAM|VRT|INV|PLN|BCT|VRL|PHG|'
-               'SYN|UNA|EST|PAT|STS|GSS|HTG|HTC|ENV|CON)'
+               ' +(?!.*DNA|.*RNA)([A-Z]{3})'
```

Comment:
Yes, we should raise an exception. if you put simply `([A-Z]{3})` no exception will be raised, cause DNA or RNA satisfy the regular expression.